libevmasm/GasMeter.h

```diff
@@ -47,37 +47,71 @@ namespace GasCosts
 	static unsigned const tier5Gas = 10;
 	static unsigned const tier6Gas = 20;
 	static unsigned const tier7Gas = 0;
-	inline unsigned extCodeGas(langutil::EVMVersion _evmVersion)
-	{
-		return _evmVersion >= langutil::EVMVersion::tangerineWhistle() ? 700 : 20;
-	}
-	inline unsigned balanceGas(langutil::EVMVersion _evmVersion)
-	{
-		if (_evmVersion >= langutil::EVMVersion::istanbul())
-			return 700;
-		else if (_evmVersion >= langutil::EVMVersion::tangerineWhistle())
-			return 400;
-		else
-			return 20;
-	}
 	static unsigned const expGas = 10;
 	inline unsigned expByteGas(langutil::EVMVersion _evmVersion)
 	{
 		return _evmVersion >= langutil::EVMVersion::spuriousDragon() ? 50 : 10;
 	}
 	static unsigned const keccak256Gas = 30;
 	static unsigned const keccak256WordGas = 6;
+	/// Corresponds to COLD_SLOAD_COST from EIP 2929
+	static unsigned const coldSloadCost = 2100;
+	/// Corresponds to COLD_ACCOUNT_ACCESS_COST from EIP 2929
+	static unsigned const coldAccountAccessCost = 2600;
+	/// Corresponds to WARM_STORAGE_READ_COST from EIP 2929
+	static unsigned const warmStorageReadCost = 100;
 	inline unsigned sloadGas(langutil::EVMVersion _evmVersion)
 	{
-		if (_evmVersion >= langutil::EVMVersion::istanbul())
+		if (_evmVersion >= langutil::EVMVersion::berlin())
+			return coldSloadCost;
+		else if (_evmVersion >= langutil::EVMVersion::istanbul())
 			return 800;
 		else if (_evmVersion >= langutil::EVMVersion::tangerineWhistle())
 			return 200;
 		else
 			return 50;
 	}
+	/// Corresponds to SSTORE_SET_GAS
 	static unsigned const sstoreSetGas = 20000;
-	static unsigned const sstoreResetGas = 5000;
+	/// Corresponds to SSTORE_RESET_GAS from EIP 2929
+	static unsigned const sstoreResetGas = 5000 - coldSloadCost;
+	inline static unsigned totalSstoreSetGas(langutil::EVMVersion _evmVersion)
+	{
+		if (_evmVersion >= langutil::EVMVersion::berlin())
+			return sstoreSetGas + coldSloadCost;
+		else
+			return sstoreSetGas;
+	}
+	inline unsigned extCodeGas(langutil::EVMVersion _evmVersion)
+	{
+		if (_evmVersion >= langutil::EVMVersion::berlin())
+			return coldAccountAccessCost;
```

Comment:
I would mention on the top of the file that in EIP-2929 we do not track warm accounts/storage slots and they are always charged the worst-case (i.e. cold access).

Reply:
Added this.